src/lib.rs

```diff
@@ -275,20 +275,44 @@ impl File {
 
     /// Persist the tempfile at the specified file path.
     ///
-    /// Note: Because of how temporary file is stored, it cannot be persisted across filesystems.
+    /// NOTE: Because of how temporary file is stored, it cannot be persisted across filesystems.
     /// Also neither the file contents nor the containing directory are
     /// synchronized, so the update may not yet have reached the disk when
     /// `persist_at` returns.
     pub fn persist_at<P: AsRef<Path>>(self, path: P) -> Result<std::fs::File, Error> {
         self.inner.persist(path).map_err(Error::TempFilePersistError)
     }
+
+    /// Create a File from the given buffer.
+    /// Could be used in testing scenarios.
+    pub fn create_from_buffer<P: AsRef<Path>>(buf: &[u8], original_file_name: String, temp_dir: Option<P>) -> Result<Self, Error> {
+        let mut file = match temp_dir {
+            Some(dir) => NamedTempFile::new_in(dir).map_err(Error::Io)?,
+            None => NamedTempFile::new().map_err(Error::Io)?,
+        };
+
+        file.write(buf).map_err(Error::Io)?;
+
+        Ok(Self {
+            inner: file,
+            original_file_name: Some(original_file_name.clone()),
+            sanitized_file_name: sanitize_filename::sanitize(original_file_name),
+        })
+    }
+
+    /// Create a File with a given size.
+    /// Could be used in testing scenarios.
+    pub fn create_with_size<P: AsRef<Path>>(size_bytes: usize, original_file_name: String, temp_dir: Option<P>) -> Result<Self, Error> {
```

Comment:
Please remove this function entirely; I don't think this library needs to provide this functionality. Users can easily do this outside the crate.